doc/modules/language-guide/pages/upgrades.adoc

```diff
@@ -1,14 +1,28 @@
 = Stable variables and upgrade methods
 
-One key feature of the {IC} is the ability to persist application data using WebAssembly memory rather than a traditional database. 
-This data storage model—referred to as **orthogonal persistence**—requires the ability to preserve and migrate the application state to support application upgrades. 
-For example, if you want to deploy a new version of an application to fix an issue or add functionality, you need to ensure that any application state that needs to be preserved remains unchanged by the upgrade process.
-The orthogonal persistence model allows an application deployed as a canister to save and restore
-state from dedicated _stable_ storage that is retained across an upgrade.
+One key feature of the {IC} is its ability to persist canister smart contract state state using WebAssembly memory and globals rather than a traditional database. This means that
+that the entire state of a canister is transparently restored before and saved after each message, without explicit user instruction.
+This automatic and user-transparent preservation of state is called
+_orthogonal persistence_.
 
-For applications written in {proglang}, the language provides high-level support for preserving state using stable storage and in a way that accommodates changes to both the application data and to the {proglang} compiler used to produce the application code.
+Though convenient, orthogonal persistence poses a challenge when it comes to upgrading the code of a canister.
+Without an explicit representation of the canister's state, how does one tranfer
+any application data from the retired canister to its replacement?
 
-Support for preserving application state depends on you—as the application programmer—anticipating and indicating the data you want to retain after an upgrade. 
+Accommodating upgrades without data loss requires some new facility to
+_migrate_ a canister's crucial data to the upgraded canister.  For example, if
+you want to deploy a new version of a user-registration canister to fix an issue or add
+functionality, you need to ensure that existing registrations survive the upgrade process.
+
+The {IC}'s persistence model allows a canister to save and restore
+such data to dedicated _stable memory_ that, unlike ordinary canister
+memory, is retained across an upgrade, allowing a canister to
+transfer data in bulk to its replacement canister.
+
+For applications written in {proglang}, the language provides high-level support for preserving state that leverages {IC} stable memory.
+This higher-level feature, called _stable storage_, is designed to accommodate changes to both the application data and to the {proglang} compiler used to produce the application code.
+
+Utilizing stable storage depends on you—as the application programmer—anticipating and indicating the data you want to retain after an upgrade.
```

Comment:
I don't know how this will be rendered but do we want spaces before the first `—` and after the second?

Similar to the above, formatting of these paragraphs make this a bit difficult to read in the source code. If you enable wrapping in your editor maybe it will make it easy to work on long lines without manually splitting them into shorter lines?

Reply:
Sure

Reply:
Done